torch/csrc/utils/python_numbers.h

```diff
@@ -126,3 +126,16 @@ inline double THPUtils_unpackDouble(PyObject* obj) {
   }
   return value;
 }
+
+inline std::complex<double> THPUtils_unpackComplexDouble(PyObject *obj) {
+  if (!PyComplex_Check(obj)) {
```

Comment:
I don't think `PyComplex_Check` sets the Python error state, so you're not allowed to throw `python_error` here.

Reply:
I edited your patch to not do this, and just allow `__complex__`